packages/amplify-provider-awscloudformation/src/initializer.ts

```diff
@@ -165,13 +173,37 @@ function cloneCLIJSONForNewEnvironment(context) {
 export async function onInitSuccessful(context) {
   configurationManager.onInitSuccessful(context);
   if (context.exeInfo.isNewEnv) {
+    if (FeatureFlags.getBoolean('overrides.project')) {
+      await storeRootStackTemplate(context);
+    }
     context = await storeCurrentCloudBackend(context);
     await storeArtifactsForAmplifyService(context);
     await uploadHooksDirectory(context);
   }
   return context;
 }
 
+export const storeRootStackTemplate = async (context: $TSContext, template?: Template) => {
+  // generate template again as the folder structure was not created when root stack was initiaized
+  if (template === undefined) {
```

Comment:
when would this be the case?

Reply:
one is init case -> no template is there since we are generating it

one is push case -> as we are adding nested resources on root stack

Reply:
so template is always undefind? Both in init and push? Then why do we need the condition?

Reply:
For push template is defined with nested stack resources as we are adding in formNestedStack function

for init its not defined yet , we ned to create it using transform